community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -53,6 +57,15 @@
      */
     Statement acquireStatement();
 
+    /**
+     * Create unique index which will be used to support uniqueness constraint.
+     *
+     * @param schema schema to create unique index for.
+     * @param provider
+     * @return IndexDescriptor for the index to be created.
+     */
+    IndexDescriptor indexUniqueCreate( SchemaDescriptor schema, Optional<String> provider );
```

Comment:
The problem is that `CreateIndexCreator` creates the unique index in a separate transaction which it starts via the kernel API, but we don't want to expose this method on the kernel API, because this should be the only caller. 
Therefore we do the cast to `KernelTransaction` in `CreateIndexCreator`. If we only have the method in `Operations` we're going to need more casting and hacking.

Reply:
Hmm, yeah! Cleaning up `ConstraintIndexCreator` might be a good idea.
Until then, what do you think about simply inlining the method from `Operations` into `ConstraintIndexCreator`? We can then just cast to `KernelTransactionImplementation` instead (like we did before) and not add this extra method to interface. Better or worse?
```
public IndexDescriptor createConstraintIndex( final SchemaDescriptor schema, Optional<String> provider )
{
    try ( Session session = kernelSupplier.get().beginSession( AUTH_DISABLED );
          Transaction transaction = session.beginTransaction( Transaction.Type.implicit ) )
    {
        IndexProvider.Descriptor providerDescriptor = indexingService.indexProviderForNameOrDefault( provider );
        IndexDescriptor index =
                IndexDescriptorFactory.uniqueForSchema( schema,
                        Optional.empty(),
                        providerDescriptor );
        ((KernelTransactionImplementation) transaction).txState().indexDoAdd( index );
        transaction.success();
        return index;
    }
    catch ( TransactionFailureException e )
    {
        throw new RuntimeException( e );
    }
}
```